scripts/generate-manifest.js

```diff
@@ -186,7 +186,23 @@ function manifestLayerV6(data, hostname, opts) {
 }
 
 function getDefaultFormat(emsFormats) {
-  return emsFormats.filter(format => format.default)[0];
+  return emsFormats.find(format => format.default);
+}
+
+function getIdsFromFile(dataDir, file, fields) {
+  const json = JSON.parse(readFileSync(`${dataDir}/${file}`, 'utf8'));
+  const features = json.features || json.objects.data.geometries;
+  const fieldMap = {};
+  for (const {name} of fields) {
+    fieldMap[name] = new Set(); // Probably unnecessary but ensures unique ids
```

Comment:
OK to have this here. 

tangent: would it make sense to add a unit tests so that every field of `"type": "id"` is a primary-key?

Reply:
Good idea. I added a unit test for this, but immediately ran into issues with legacy layers. So I demoted the unit test to only run when the [`EMS_STRICT_TEST` variable is set](https://github.com/elastic/ems-file-service#usage). We can fix the legacy layers in a separate PR. 

Once we fix those layers we can promote the unit test and probably remove this `Set`.